cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/compiler_adapt.hpp

```diff
@@ -83,6 +88,47 @@ ONEDAL_FORCEINLINE std::int32_t ONEDAL_popcnt64(std::uint64_t a) {
 #endif
 }
 
+#ifdef __ARM_ARCH
+template <>
+ONEDAL_FORCEINLINE std::int32_t ONEDAL_lzcnt_u32<dal::backend::cpu_dispatch_sve>(std::uint32_t a) {
+    if (a == 0)
+        return 32;
+    std::uint32_t one_bit = 0x80000000; // binary: 1000 0000 0000 0000 0000 0000 0000 0000
+    std::int32_t bit_pos = 0;
+    while ((a & one_bit) == 0) {
+        bit_pos++;
+        one_bit >>= 1;
+    }
+    return bit_pos;
```

Comment:
I would consider using [compiler specific builtins](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) wherever it is possible. In case of GCC/Clang it would be [`__builtin_clz`](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-_005f_005fbuiltin_005fclz)